vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/transformer/AbstractFetchingUrlsTableTransformer.java

```diff
@@ -42,6 +47,12 @@
     private WebApplicationConfiguration webApplicationConfiguration;
     private HttpRedirectsProvider httpRedirectsProvider;
     private boolean filterRedirects;
+    private final Supplier<URI> mainApplicationPageUriSupplier = Suppliers.memoize(() ->
+    {
+        URI mainApplicationPageUrl = webApplicationConfiguration.getMainApplicationPageUrl();
```

Comment:
Could you please refactor to check in one place?
We have a similar check here: https://github.com/vividus-framework/vividus/blob/master/vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/PageSteps.java#L92-L99

Reply:
resolve this